applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/eigenfrequency_response_function_utility.h

```diff
@@ -64,327 +64,434 @@ namespace Kratos
+    {
+        KRATOS_ERROR_IF_NOT(rResponseSettings.Has("weighting_factors")) << "No weighting factors defined for given eigenfrequency response!" << std::endl;
+        KRATOS_ERROR_IF_NOT(rResponseSettings["weighting_factors"].size() == mTracedEigenfrequencyIds.size()) << "The number of chosen eigenvalues does not fit to the number of weighting factors!" << std::endl;
+
+        mWeightingFactors.resize(rResponseSettings["weighting_factors"].size(),false);
+
+        // Read weighting factors and sum them up
+        double test_sum_weight_facs = 0.0;
+        for(std::size_t i = 0; i < mWeightingFactors.size(); i++)
+        {
+            mWeightingFactors[i] = rResponseSettings["weighting_factors"][i].GetDouble();
+            test_sum_weight_facs += mWeightingFactors[i];
+        }
+
+        // Check the weighting factors and modify them for the case that their sum is unequal to one
+        if(std::abs(test_sum_weight_facs - 1.0) > 1e-12)
+        {
+            for(std::size_t i = 0; i < mTracedEigenfrequencyIds.size(); i++)
+                mWeightingFactors[i] /= test_sum_weight_facs;
+
+            std::cout << "> The sum of the chosen weighting factors is unequal to one. A scaling process was executed for them!" << std::endl;
+        }
+    }
+
+    // --------------------------------------------------------------------------
+    void UseDefaultWeight()
+    {
+        mWeightingFactors.resize(1,false);
+        mWeightingFactors[0] = 1.0;
+    }
+
+    // --------------------------------------------------------------------------
+    void CheckIfAllNecessaryEigenvaluesAreComputed()
+    {
+        const int num_of_computed_eigenvalues = (mrModelPart.GetProcessInfo()[EIGENVALUE_VECTOR]).size();
+        const int max_required_eigenfrequency = *std::max_element(mTracedEigenfrequencyIds.begin(), mTracedEigenfrequencyIds.end());
+        KRATOS_ERROR_IF(max_required_eigenfrequency > num_of_computed_eigenvalues) << "The following Eigenfrequency shall be traced but was not computed by the eigenvalue analysies: " << max_required_eigenfrequency << std::endl;
+    }
+
+    // --------------------------------------------------------------------------
+    void CalculateCompleteResponseDerivativeByFiniteDifferencing()
+    {
+        ProcessInfo &CurrentProcessInfo = mrModelPart.GetProcessInfo();
+
+        // Predetermine all necessary eigenvalues and prefactors for gradient calculation
+        const unsigned int num_of_traced_eigenfrequencies = mTracedEigenfrequencyIds.size();
+        Vector traced_eigenvalues(num_of_traced_eigenfrequencies,0.0);
+        Vector gradient_prefactors(num_of_traced_eigenfrequencies,0.0);
+        for(std::size_t i = 0; i < num_of_traced_eigenfrequencies; i++)
+        {
+            traced_eigenvalues[i] = GetEigenvalue(mTracedEigenfrequencyIds[i]);
+            gradient_prefactors[i] = 1 / (2*Globals::Pi * std::sqrt(traced_eigenvalues[i]));
+        }
```

Comment:
Maybe I missed something be should the pre-factor not be '1/(2*2*pi*sqrt(eigenvalue))'? One 2 from the conversion from eigenvalue to eigenfrequency and one 2 from the differentiation of the square root when using the chain rule?

Reply:
Of course, It missed that. Thanks for the hint!

Reply:
You missed to correct this bug.